pyomo/contrib/pynumero/interfaces/pyomo_nlp.py

```diff
@@ -1018,3 +1117,283 @@ def load_state_into_pyomo(self, bound_multipliers=None):
+
+        # Hw_eq(x)
+        eq_hess = None
+        if self._ex_model.n_equality_constraints() > 0:
+            eq_hess = self._ex_model.evaluate_hessian_equality_constraints()
+            if self._eq_hess_jcol is None:
+                # first time through, let's also check that it is lower triangular
+                if np.any(eq_hess.row - eq_hess.col < 0):
+                    raise ValueError('ExternalGreyBoxModel must return lower '
+                                     'triangular portion of the Hessian only')
+
+                # The first time through, we won't have created the
+                # mapping of external primals ('u') to the full space
+                # primals ('x')
+                self._eq_hess_irow = row = self._inputs_to_primals_map[eq_hess.row]
+                self._eq_hess_jcol = col = self._inputs_to_primals_map[eq_hess.col]
+
+                # mapping may have made this not lower triangular
+                mask = col > row
+                if np.any(mask):
+                    temp = row[mask].copy()
+                    row[mask] = col[mask]
+                    col[mask] = temp
+
+            data_list.append(eq_hess.data)
+            irow_list.append(self._eq_hess_irow)
+            jcol_list.append(self._eq_hess_jcol)
+            # # map the columns from the inputs "u" back to the full primals "x"
+            # eq_hess = coo_matrix(
+            #     (eq_hess.data, (self._eq_hess_irow, self._eq_hess_jcol)),
+            #     (self._n_primals, self._n_primals))
+
+        if self._ex_model.n_outputs() > 0:
+            output_hess = self._ex_model.evaluate_hessian_outputs()
+            if self._output_hess_irow is None:
+                # first time through, let's also check that it is lower triangular
+                if np.any(output_hess.row - output_hess.col < 0):
+                    raise ValueError('ExternalGreyBoxModel must return lower '
+                                     'triangular portion of the Hessian only')
+
+                # The first time through, we won't have created the
+                # mapping of external outputs ('o') to the full space
+                # primals ('x')
+                self._output_hess_irow = row = self._inputs_to_primals_map[output_hess.row]
+                self._output_hess_jcol = col = self._inputs_to_primals_map[output_hess.col]
+
+                # mapping may have made this not lower triangular
+                mask = col > row
+                if np.any(mask):
+                    temp = row[mask].copy()
+                    row[mask] = col[mask]
+                    col[mask] = temp
+
+            data_list.append(output_hess.data)
+            irow_list.append(self._output_hess_irow)
+            jcol_list.append(self._output_hess_jcol)
+            # outputs_hess = coo_matrix(
+            #     (outputs_hess.data, (self._outputs_hess_irow, self._outputs_hess_jcol)),
+            #     (self._n_primals, self._n_primals))
```

Comment:
Can this be removed?

Reply:
yep. And I changed the np.any stuff above.